internal/translator/data/rules_test.go

```diff
@@ -0,0 +1,232 @@
+			Pattern:        "VERB,ADV,ADJ",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "M",
+				},
+				{
+					Tag:  "ADV",
+					Type: "G",
+				},
+				{
+					Tag:  "ADJ",
+					Type: "O",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 1).Lemma}}",
+				},
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("find rule with success when the value exists", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Found rules %v", rule)
+	})
+
+	t.Run("get and error finding non existing rule", func(t *testing.T) {
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "ADJ,ADJ,ADV")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		if len(rule) == 0 {
+			t.Logf("pattern ADJ,ADJ,ADV doesn't exist")
+		}
+	})
+
+	t.Run("delete a group of rules with success when the rules exist", func(t *testing.T) {
+		error := helper.DeleteMany(context.Background(), "espaol", "kaqchikel", "VERB,ADV,ADJ")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("Rules deleted")
+	})
+
+	t.Run("add short new rule with success when the strucutre is valid", func(t *testing.T) {
+		newRule := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "VERB",
+			Details: []RuleDetail{
+				{
+					Tag:  "VERB",
+					Type: "M",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		newID, error := helper.NewRule(context.Background(), newRule)
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		t.Logf("New rule created with id %s", newID.Hex())
+	})
+
+	t.Run("update one rule with success when the rule exists", func(t *testing.T) {
+		filter, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "VERB")
+
+		if error != nil {
+			t.Fatal(error)
+		}
+
+		sameRuleChange := Rule{
+			SourceLanguage: "espaol",
+			TargetLanguage: "kaqchikel",
+			Pattern:        "ADJ",
+			Details: []RuleDetail{
+				{
+					Tag:  "ADJ",
+					Type: "",
+				},
+			},
+			Output: []RuleOutput{
+				{
+					"type":  "direct-translation",
+					"value": "{{(index .Words 0).Lemma}}",
+				},
+			},
+		}
+
+		erroru := helper.UpdateOne(context.Background(), filter[0], sameRuleChange)
+
+		if erroru != nil {
+			t.Fatal(erroru)
+		}
+
+		rule, error := helper.FindRuleByPattern(context.Background(), "espaol", "kaqchikel", "ADJ")
+
+		t.Logf("Rule updated %v", rule)
+	})
+
+	t.Run("delete one rule with success when the rule exists", func(t *testing.T) {
```

Comment:
What if the single rule does not exist.

Reply:
the array len is 0, test was changed in code

Reply:
Ok, but remember one test only should test one thing. It means a `success` and `error` cases should be two tests. 

SRP

Reply:
ok I will keep that in mind for future testing